exp/controllers/awsmachinepool_controller.go

```diff
@@ -607,7 +623,7 @@ func machinePoolToInfrastructureMapFunc(gvk schema.GroupVersionKind) handler.Map
 	}
 }
 
-func (r *AWSMachinePoolReconciler) getInfraCluster(ctx context.Context, log *logger.Logger, cluster *clusterv1.Cluster, awsMachinePool *expinfrav1.AWSMachinePool) (scope.EC2Scope, error) {
+func (r *AWSMachinePoolReconciler) getInfraCluster(ctx context.Context, log *logger.Logger, cluster *clusterv1.Cluster, awsMachinePool *expinfrav1.AWSMachinePool) (scope.EC2Scope, scope.S3Scope, error) {
```

Comment:
It looks like there is no case where the first and second returned values are different. They are always both the same value, either `nil` or `clusterScope` or `managedControlplaneScope`. If both always have the same value, couldn't we just return one instead of two values?

Reply:
See above comment regarding interfaces